### DISCUSSION TITLE
partial void OnXXXXXXChanged(string value) in viewmodel error

### DISCUSSION THREAD

**Discussion by Gimmy-Ateikon** [CommunityToolkit/dotnet#528] [Q&A] [answered]
I state that I am a neophyte. 
I am attempting to implement a class binding using the toolkit 8.0 in .Net Maui ; everything seems to work apart from generating 'partial void OnXXXXXXChanged(string value)'. 
In the viewModel gives me this error:
![errro](https://user-images.githubusercontent.com/116343721/206923479-81528ae6-9904-45d2-8300-e5d02a6a86e3.png)
Can someone help me ?

**Comment by Gimmy-Ateikon** [comment]
This is my viewModel
==============

```csharp
namespace Alina.ViewModel
{
    [INotifyPropertyChanged]
    public partial class BindingClassToolkitViewModel
    {

        public BindingClassToolkitViewModel()
        {
        }

        public ArticoloModel ArticoloModel { get; set; } = new ArticoloModel();

      
        [RelayCommand]
        public async Task AddArticolo()
        {
            // Add to database !
            ArticoloModel.ClearFields();
        }

// HERE I HAVE THE ERROR

        // Alla pressione di ogni tasto scatta l'evento.
        // esiste un evento per ogni pubblica variabile: OnXxxxxChanged
        partial void OnCodiceChanged(string value)
        {
            App.Current.MainPage.DisplayAlert("Changed", value, "OK");
        }

    }
}
```

**Comment by Gimmy-Ateikon** [comment]
My view
======


<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:local="clr-namespace:Alina.ViewModel"
             x:Class="Alina.View.BindingClassToolkitPage"
             Title="BindingClassToolkitPage">
    
    <ContentPage.BindingContext>
        <local:BindingClassToolkitViewModel/>
    </ContentPage.BindingContext>

    <ScrollView HorizontalOptions="CenterAndExpand">
        <VerticalStackLayout Padding="40" Spacing="10">

            <Label Text="Codice" />
            <Entry Text="{Binding ArticoloModel.Codice}" BackgroundColor="AliceBlue"  />
            <Label Text="Descrizione" />
            <Entry Text="{Binding ArticoloModel.Descrizione}"  BackgroundColor="AliceBlue"/>
            <Label Text="Tipo" />
            <Entry Text="{Binding ArticoloModel.Tipo}" BackgroundColor="AliceBlue" />
            <Label Text="{Binding ArticoloModel.FullArticle}" />

            <Button Text="Aggiungo Articolo" 
                Command="{Binding AddArticoloCommand }" 
                HorizontalOptions="Center"/>
        </VerticalStackLayout>
    </ScrollView>

</ContentPage>

**Comment by Gimmy-Ateikon** [comment]
my model
=======


using CommunityToolkit.Mvvm.ComponentModel;
using System.ComponentModel;

namespace Alina.Model

{
    [INotifyPropertyChanged]
    public partial class ArticoloModel
    {

        [ObservableProperty]
        [NotifyPropertyChangedFor(nameof(FullArticle))]
        string codice ;

        [ObservableProperty]
        [NotifyPropertyChangedFor(nameof(FullArticle))]
        string descrizione;

        [ObservableProperty]
        [NotifyPropertyChangedFor(nameof(FullArticle))]
        string tipo;

        public string FullArticle => $"{Codice} - {Descrizione}";
        
        public void ClearFields()
        {
            Codice = string.Empty;
            Descrizione = string.Empty;
            Tipo = string.Empty;        }

    }
}

**Comment by Sergio0694** [comment, marked as answer]
Ciao! The issue is that you simply don't have a generated "Codice" property there. The partial "On___Changed" methods are generated for properties you declare via `[ObservableProperty]`, and you don't have such a property in `BindingClassToolkitViewModel` (you have one in `ArticoloModel`).

Also, just a side note: avoid using `[INotifyPropertyChanged]` if you can inherit from `[ObservableObject]` instead.

**Reply by Gimmy-Ateikon** [reply]
Hello. 
Thank you for your reply. 
The "codice", as you noticed, is in my model. 
I deduce that, if I Binding a class and not properies in viewmodel I can not use automatic events? 
How can I overcome this?  
Thank you for your patience. 

P.s. 
Thanks for the suggestion about [INotifyPropertyChanged]/[ObservableObject]

P.p.s
Vedo che sei italiano. Qui è ammesso scrivere in italiano
